### PR TITLE
Fix wrong context from modal

### LINKED ISSUE #2229: Fix wrong navigation scope from modal
- Fix issue #2211 : LuigiClient linkManager wrong navigation scope in modal  (PR #2221)

Little refactoring on `buildPath` and `getSubPath` functions at App.html
- Make it stateless. Reduce the usage of global variable _(params, navigationPath, pathParams, currentNode)_. 
- Make those arguments as optional pass to function.

### PR DESCRIPTION
**Pull request by stanleychh**
**Description**

Changes proposed in this pull request:

- Refactor `buildPath` and `getSubPath` at App.html
- Fix issue #2211 : LuigiClient linkManager wrong navigation scope in modal

**Related issue(s)**
Fixes #2229

### CODE REVIEW

**Review by ndricimrr — COMMENTED** [review]
So far so good. But there are currently some tests failing after merge from master. Would be good to have a look at that, if it can be fixed.

**Comment by stanleychh** [comment]
@ndricimrr Thanks for reviewing. Delay the DOM rendering fixed the broken test.

**Review by ndricimrr — APPROVED** [review]
Re-tested. Not able to find and issues. LGTM 👍

File: core/src/App.html [hunk 1]
```diff
@@ -769,7 +792,7 @@
   let alerts = [];
 
   const getAlertWithId = (alertQueue, id) => {
-    if (!alertQueue || !alertQueue.length) return;
+    if (!alertQueue || !alertQueue.length > 0) return;
```
Comment (hardl) [inline core/src/App.html:795]: Wrong check... in this context it always leads to the correct result, but be aware that 
!x > 0 is not the same as !(x > 0) for negative x!

File: core/src/App.html [hunk 2]
```diff
@@ -1088,7 +1111,7 @@
 
     EventListenerHelpers.addEventListener('popstate', async e => {
       const alertQueue = alerts;
-      if (!alertQueue || !alertQueue.length) return;
+      if (!alertQueue || !alertQueue.length > 0) return;
```
Comment (hardl) [inline core/src/App.html:1114]: Wrong check... in this context it always leads to the correct result, but be aware that 
!x > 0 is not the same as !(x > 0) for negative x!

File: core/src/App.html (hunk 1 above)
Comment (stanleychh) [reply core/src/App.html]: Good point! Thanks for reviewing it.

**Pull request merged**